sitestats/sitestats-impl/src/java/org/sakaiproject/sitestats/impl/report/ReportManagerImpl.java

```diff
@@ -119,9 +120,22 @@ public class ReportManagerImpl extends HibernateDaoSupport implements ReportMana
 	private ReportFormattedParams	formattedParams	= new ReportFormattedParamsImpl();
 
 	/** FOP */
-	private FopFactory				fopFactory		= FopFactory.newInstance();
-	private Templates				cachedXmlFoXSLT	= null;
-	private static final String		XML_FO_XSL_FILE	= "xmlReportToFo.xsl";
+	private static final FopFactory fopFactory;
+	private Templates cachedXmlFoXSLT = null;
+	private static final String XML_FO_XSL_FILE = "xmlReportToFo.xsl";
+
+	// Static initialization block for the thread-safe FopFactory
+	static {
+		// Create a factory with default configuration
+		// We're using a safe, absolute URI that doesn't depend on current directory
+		try {
+			fopFactory = FopFactory.newInstance(new File(System.getProperty("java.io.tmpdir")).toURI());
+		} catch (Exception e) {
+			// Log error but allow class to load - will throw exception at runtime if used
+			log.error("Failed to initialize FOP factory", e);
+			throw new RuntimeException("Failed to initialize FOP factory", e);
+		}
+	}
```

Comment:
it would be preferredif the FopFactory initialization can be done in a constructor, there likely is no need to do it statically as this is already a singleton.

Reply:
ok done